src/main/java/com/marklogic/kafka/connect/sink/MarkLogicSinkTask.java

```diff
@@ -154,7 +155,12 @@ public void put(final Collection<SinkRecord> records) {
 					logger.debug("Processing record value {} in topic {}", record.value(), record.topic());
 				}
 				if (record.value() != null) {
-					writeBatcher.add(sinkRecordConverter.convert(record));
+					try {
+						writeBatcher.add(sinkRecordConverter.convert(record));
+					}
+					catch (IOException e) {
```

Comment:
See my comment below - why catch and log but not rethrow only an IOException here?

Reply:
This will be addressed in the follow-up refactor issue.